resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/repository/ReactivePersistenceLayer.java

```diff
@@ -590,68 +650,20 @@ private void saveSerialisedFormat(final String serialisedFormat, final LeafResou
-    // * type repository:     there exist entries for this leaf's type
-    // * ser.For. repository: there exist entries for this leaf's serialisedFormat
-    //
-    // As long as this is called for every new resource created and added to the resource-service,
-    // this guarantees consistency between persistence and resource-service
-    @Override
-    public void addResource(final LeafResource leafResource) {
-        LOGGER.debug("Persistence update for new leafResource with resourceId {}", leafResource.getId());
-        // Update resources repository
-        Optional<Resource> firstCompleteAncestor = Optional.empty();
-        // Find the first direct ancestor (grand*parent) that is persisted and complete
-        // This informs us how much of the tree needs updating
-        for (Resource ancestor = leafResource; ancestor instanceof ChildResource; ancestor = ((ChildResource) ancestor).getParent()) {
-            if (isResourceIdComplete(ancestor.getId()) && isResourceIdPersisted(ancestor.getId())) {
-                firstCompleteAncestor = Optional.of(ancestor);
-                break;
-            }
-        }
-        // Treat this ancestor as the 'root' resource for a putResourcesById (if such an ancestor was found)
-        // Otherwise we can use this leafResource and mark just that as complete
-        // Useful for when there is persisted information about the resource's type
-        Resource effectiveRoot = firstCompleteAncestor.orElse(leafResource);
-        LOGGER.debug("Adding resource '{}' with effective root '{}'", leafResource.getId(), effectiveRoot.getId());
-        saveResourceWithIncompleteParents(leafResource);
-
-        // Update type repository if there is already a complete set of info
-        String type = leafResource.getType();
-        if (isTypeComplete(type)) {
-            saveType(type, leafResource);
-        } else {
-            LOGGER.debug("Skipping add for type {} as entry not present (partial store would be incomplete)", type);
-        }
-
-        // Update serialisedFormat repository if there is already a complete set of info
-        String serialisedFormat = leafResource.getSerialisedFormat();
-        if (isSerialisedFormatComplete(serialisedFormat)) {
-            saveSerialisedFormat(serialisedFormat, leafResource);
-        } else {
-            LOGGER.debug("Skipping add for serialised format {} as entry not present (partial store would be incomplete)", serialisedFormat);
-        }
+        return Flow.lazyCompletionStageFlow(() -> isSerialisedFormatComplete(serialisedFormat)
+                .thenCompose((Boolean result) -> {
+                    if (result.equals(Boolean.FALSE)) {
+                        return completenessRepository.futureSave(EntityType.FORMAT, serialisedFormat);
+                    } else {
+                        return CompletableFuture.completedFuture(null);
+                    }
+                })
```

Comment:
A couple of these variables like this could be more-descriptively named
```suggestion
                .thenCompose((Boolean formatAlreadyComplete) -> {
                    if (formatAlreadyComplete.equals(Boolean.FALSE)) {
                        return completenessRepository.futureSave(EntityType.FORMAT, serialisedFormat);
                    } else {
                        return CompletableFuture.completedFuture(null);
                    }
                })
```
It'd also be nice to not use `Boolean`s and just use `bool`s if the change isn't too tricky

Reply:
Happy to change the variable names to something more descriptive. 
Changing the `Boolean` to a `bool` is not going to be easy as the `isSomethingComplete()` methods all return `CompletableFuture<Boolean>` which is created from a `Mono<Boolean>` when checking a repository for an instance (this returns a `Mono<Entity>` value).